### ISSUE TITLE
infinite recursion prevention

### ISSUE THREAD

**Issue by bglusman** [alexdesousa/ayesql#42]
If a query named `:x` contains a reference to `:x` as a parameter inside it, it will trigger infinite recursion.... this should probably be detected and prevented with a clear error for improved developer experience

**Comment by alexdesousa** [comment]
Certainly, this would help developer experience! When I was building an example for #41 I run into this issue. Basically, I had a fragment with a the same name as the parameter:

```sql
-- name: limit
LIMIT :limit
```

and I fixed it with:

```sql
-- name: limit
LIMIT :rows
```

This would be a nice to have in the future.

**Comment by bglusman** [comment]
Yeah, that's exactly the situation I hit it in, maybe not limit but same idea.  While I was making the other PR I was thinking about how I'd do this one too, it didn't look simple, but I also didn't get that far into understanding how the templates were compiled.  If you get around to it, awesome, if you'd like me to take a crack at, LMK, but a little high level guidance for what you think best approach might be would be appreciated.  Thanks!
